src/overlays/actors/ovl_En_Baba/z_en_baba.c

```diff
@@ -311,212 +334,219 @@ void func_80BA8DF4(EnBaba* this, GlobalContext* globalCtx) {
     return true;
 }
 
-s32 func_80BA9110(EnBaba* this, GlobalContext* globalCtx, ScheduleResult* arg2) {
-    s32 ret;
+s32 EnBaba_ProcessScheduleResult(EnBaba* this, GlobalContext* globalCtx, ScheduleResult* scheduleResult) {
+    s32 success;
 
-    switch (arg2->result) {
+    switch (scheduleResult->result) {
         default:
-            ret = false;
+            success = false;
             break;
 
-        case 2:
-            ret = func_80BA8F88(this, globalCtx, arg2);
+        case BABA_SCHEDULE_RESULT_FOLLOW_TIME_PATH:
+            success = EnBaba_InitTimePath(this, globalCtx, scheduleResult);
             break;
 
-        case 1:
-            ret = true;
+        case BABA_SCHEDULE_RESULT_KNOCKED_OVER:
+            success = true;
             break;
     }
-    return ret;
+    return success;
 }
 
-s32 func_80BA9160(EnBaba* this, GlobalContext* globalCtx) {
-    f32 sp7C[265];
-    Vec3f sp70;
-    Vec3f sp64;
-    Vec3f sp58;
-    s32 sp54 = 0;
-    s32 sp50 = 0;
+s32 EnBaba_FollowTimePath(EnBaba* this, GlobalContext* globalCtx) {
+    f32 weightArray[265];
+    Vec3f worldPos;
+    Vec3f timePathPointNew; // used to compute yaw after update
+    Vec3f timePathPoint;    // used in setup and to store a backup before update
+    s32 timePathElapsedTime = 0;
+    s32 timePathWaypoint = 0;
     s32 pad;
 
-    func_8013AF00(sp7C, 3, this->unk_410->count + 3);
+    SubS_TimePathing_FillWeightArray(weightArray, 3, this->timePath->count + 3);
 
-    if (this->unk_438 == 0) {
-        sp58 = gZeroVec3f;
+    if (!this->timePathIsSetup) {
+        timePathPoint = gZeroVec3f;
 
-        func_8013B6B0(this->unk_410, &this->unk_420, &this->unk_430, this->unk_428, this->unk_424, &this->unk_42C, sp7C,
-                      &sp58, this->unk_436);
-        func_8013B878(globalCtx, this->unk_410, this->unk_42C, &sp58);
-        this->actor.world.pos.y = sp58.y;
-        this->unk_438 = 1;
+        SubS_TimePathing_Update(this->timePath, &this->timePathWeightVal, &this->timePathElapsedTime,
+                                this->timePathWaypointTime, this->timePathTotalTime, &this->timePathWaypoint,
+                                weightArray, &timePathPoint, this->timePathTimeSpeed);
+        SubS_TimePathing_ComputeInitialY(globalCtx, this->timePath, this->timePathWaypoint, &timePathPoint);
+        this->actor.world.pos.y = timePathPoint.y;
+        this->timePathIsSetup = true;
     } else {
-        sp58 = this->unk_414;
+        timePathPoint = this->timePathPoint;
     }
 
-    this->actor.world.pos.x = sp58.x;
-    this->actor.world.pos.z = sp58.z;
+    this->actor.world.pos.x = timePathPoint.x;
+    this->actor.world.pos.z = timePathPoint.z;
 
     if (SubS_InCsMode(globalCtx)) {
-        sp54 = this->unk_430;
-        sp50 = this->unk_42C;
-        sp58 = this->actor.world.pos;
+        timePathElapsedTime = this->timePathElapsedTime;
+        timePathWaypoint = this->timePathWaypoint;
+        timePathPoint = this->actor.world.pos;
```

Comment:
```suggestion
        prevTimePathElapsedTime = this->timePathElapsedTime;
        prevTimePathWaypoint = this->timePathWaypoint;
        timePathPoint = this->actor.world.pos;
```
because these variables are just storing values if in a cutscene, I think `prev` is an indication that something is being stored and restored later on. I didn't add `prev` to `timePathPoint` since that's used in other ways as well. Thoughts?